modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -531,4 +534,13 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
             }
         }
     }
+
+    protected HttpContentDecompressor createDecompressor() {
```

Comment:
Why allow the decompression system to be modified?

Reply:
I was able to define a conditional decompressor in https://github.com/opensearch-project/security/pull/3418 which relied on context from earlier in the pipeline. The context attribute to determine whether to decompress/keep compressed was defined in the `header_verifier` step in the same PR so both steps of the pipeline needed to be extended.

Reply:
I was able to get an ChannelInboundHandlerAdapter to properly process a request and skip the pipeline - I've removed this.